great_expectations/checkpoint/actions.py

```diff
@@ -464,8 +464,13 @@ def run(self, checkpoint_result: CheckpointResult, action_context: ActionContext
             data_docs_pages=data_docs_pages,
         )
         # this will actually sent the POST request to the Microsoft Teams webapp server
+        webhook = self._substitute_config_str_if_needed(self.teams_webhook)
```

Comment:
Method returns a nullable type but we're always guaranteed this

Reply:
We can tighten up the typing on this to make this check unnecessary by using a generic (let's get to that in a followup)